features/collection/persistence-rrd/src/main/java/org/opennms/netmgt/collection/persistence/rrd/RrdPersistOperationBuilder.java

```diff
@@ -211,7 +222,9 @@ public void commit() throws PersistException {
             List<RrdDataSource> dataSources = getDataSources();
             if (dataSources != null && dataSources.size() > 0) {
                 createRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, getRepository().getStep(), dataSources, getRepository().getRraList(), m_metaData);
-                updateRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, m_timeKeeper.getCurrentTime(), getValues());
+                long rrdTime = Boolean.getBoolean("org.opennms.netmgt.collection.persistence.rrd.useCollectionStartTime")
+                    ? m_collectionTime : m_timeKeeper.getCurrentTime();
```

Comment:
There are a lot of CollectionResource objects that would need updating. I was thinking we could wrap those objects instead.

The code in CollectableService could be replaced with something like:

```java
if (Boolean.getBoolean("org.opennms.netmgt.collection.persistence.useCollectionStartTime")) {
  final ConstantTimeKeeper timeKeeper = new ConstantTimeKeeper(new Date(0));
  // Wrap the persister visitor such that calls to CollectionResource.getTimeKeeper() return the given timeKeeper
  persister = wrapResourcesWithTimekeeper(persister, timeKeeper);           
}
```

where `wrapResourcesWithTimekeeper` is defined as:
```java
private static CollectionSetVisitor wrapResourcesWithTimekeeper(CollectionSetVisitor visitor, TimeKeeper timeKeeper) {
        // Wrap the given visitor and intercept the calls to visit the resources
        final CollectionSetVisitor wrappedVisitor = new CollectionSetVisitorWrapper(visitor) {
            private CollectionResource wrappedResource;

            @Override
            public void visitResource(CollectionResource resource) {
                // Wrap the given resource and return the custom timekeeper
                wrappedResource = new CollectionResourceWrapper(resource) {
                    @Override
                    public TimeKeeper getTimeKeeper() {
                        return timeKeeper;
                    }
                };
                getDelegate().visitResource(wrappedResource);
            }

            @Override
            public void completeResource(CollectionResource resource) {
                getDelegate().completeResource(wrappedResource);
            }
        };
        return wrappedVisitor;
    }
```

`org.opennms.netmgt.collection.support.CollectionSetVisitorWrapper` already exists, we would just need to create the analogue for the CollectionResource.